mlflow/cli.py

```diff
@@ -573,6 +637,17 @@ def gc(older_than, backend_store_uri, run_ids):
         backend_store._hard_delete_run(run_id)
         click.echo("Run with ID %s has been permanently deleted." % str(run_id))
 
+    if not skip_experiments:
+        for experiment_id in experiment_ids:
+            if older_than and experiment_id not in deleted_older_experiment_ids:
+                raise MlflowException(
+                    f"Experiment {experiment_id} is not older than the required age. "
+                    f"Only runs older than {older_than} can be deleted.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
```

Comment:
I think we should perform this validation before deleting experiments:

```python
# Enusre all the experiments are hard-deleted
...


# Then, delete them
for experiment_id in experiment_ids:
    ...
```

Reply:
Current version:
```
        for experiment_id in experiment_ids:
            if older_than and experiment_id not in deleted_older_experiment_ids:
                raise MlflowException(
                    f"Experiment {experiment_id} is not older than the required age. "
                    f"Only runs older than {older_than} can be deleted.",
                    error_code=INVALID_PARAMETER_VALUE,
                )
            backend_store._hard_delete_experiment(experiment_id)
```

Suggested version:
```
        for experiment_id in experiment_ids:
            if older_than and experiment_id not in deleted_older_experiment_ids:
                raise MlflowException(
                    f"Experiment {experiment_id} is not older than the required age. "
                    f"Only runs older than {older_than} can be deleted.",
                    error_code=INVALID_PARAMETER_VALUE,
                )
         for experiment_id in experiment_ids:
            backend_store._hard_delete_experiment(experiment_id)
```

@harupy  Is that correct?

Reply:
We could do something like this:

```python
invalid_experiments = []
for experiment_id in experiment_ids:
    # append to `invalid_experiments` if this experiment can't be deleted

if invalid_experiments:
    # Throw

# Delete experiments
...
```

Reply:
@harupy  I have made the change. But, if we have a long list of `invalid_experiments`, then the error message will be very long. Is that fine?

Reply:
I think it's fine.